File: tacosort/javascript/taco.js
function taco(cat) {
  var list = cat;

  while (notSorted(list)) {
    list = shuffle(list);
  }

  console.log('Hurray! You did it!');
  console.log(list);
}

function notSorted(yet) {
  // A list is 'sorted' if the list is of integers and they are ordered from smallest to largest
  for (var i = 0; i < yet.length - 1; i++) {
    if (yet[i] > yet[i + 1]) {
      console.log('Still not sorted...');
      return true;
    }
  }

  return false;
}

function shuffle(it) {
  var list = it.slice(0); // Make a copy of the list since splice is destructive
  // Remove a random element from the list and add it back in a random position
  list.splice(getRandomNumberBetween(0, list.length - 1), 0, list.splice(getRandomNumberBetween(0, list.length - 1), 1)[0]);
  return list;
}

function getRandomNumberBetween(min, max) {
  return Math.floor(Math.random() * (max - min + 1)) + min;
}

taco([5,7,2,9,1,4]);

File: tacosort/swift/taco.swift
func taco(list: [Int]){
    var taco = list
        while taco != taco.sorted(){
        let rand1:UInt32 = arc4random_uniform(UInt32(list.count))
        let rand2:UInt32 = arc4random_uniform(UInt32(list.count))
        let meat = taco.remove(at: Int(rand1))
        taco.insert(meat, at: Int(rand2))
        print(taco)
    }
    
}
taco(list: [5,7,2,9,1,4])